README.md

```diff
@@ -68,231 +70,1017 @@ Dette er i svært korte trekk det grunnleggende du trenger å vite om React. I l
+
+Neste steg er å vise antall likes:
+
+```js
+function Likes(props) {
+  const [likes, setLikes] = React.useState(0);
+  return <div className="likes">Likes: {likes}</div>;
+}
+```
+
+Dette får den første testen vår til å kjøre. Når vi ser på websiden, ser vi at det står "Likes: 0"
+
+Del to av oppgaven består i å lage en knapp man kan trykke på, og som legger til en til antall likes.
+
+```js
+function Likes(props) {
+  const [likes, setLikes] = React.useState(0);
+  function incrementLikes() {
+    setLikes(likes + 1);
+  }
+  return (
+    <div className="likes">
+      Likes: {likes}{' '}
+      <button className="like-button" onClick={incrementLikes}>
+        💛
+      </button>
+    </div>
+  );
+}
+```
+
+Vi lager først en ny funksjon `incrementLikes`, som kaller `setLikes`-funksjonen med antall likes + 1. Du kan også sende inn en funksjon som tar imot nåværende verdi, og som returnerer oppdatert verdi:
+
+```js
+function incrementLikes() {
+  setLikes(currentLikes => currentLikes + 1);
+}
+```
+
+Man bør bruke sistnevnte om den nye verdien avhenger av den gamle verdien - for å garantere at ikke noe annet oppdaterer antall likes i mellomtiden.
+
+Det var det! Vi kan nå like bildene våre! Om det bare var en måte å la serveren vår huske det på...
+
+</details>
 
 ### Oppgave 8: Sideeffekter
 
 I oppgave 8 skal vi fokusere på å utføre forskjellige side-effekter. Side-effekter er handlinger som påvirker noe utenfor "React-verdenen" - som å kalle DOM-APIer, hente data og så videre
 
 ### 8A: Oppdater tittel
-Når man går inn på et bilde burde man oppdatere tittelen til websiden (det som står oppe i fanen). Bruk hooken `useEffect` til å oppdatere den!
 
-https://codesandbox.io/s/oppgave-8-min-forste-side-effekt-o5xlp
+Når man går inn på et bilde burde man oppdatere tittelen til websiden (det som står oppe i fanen).
+
+🏆 Bruk hooken `useEffect` til å oppdatere tittelen til å si "📷 av @brukernavn" når man går inn på en detaljside.
+
+> 💡 Du kan sette sidetittelen med å endre `document.title`
+
+<details><summary>🚨 Løsningsforslag</summary>
+En side-effekt er noe som påvirker noe utenfor React-verdenen. Det kan være å kalle DOM-APIer, hente data eller noe helt annet. I dette tilfellet vil vi oppdatere dokument-tittelen - den tekststrengen som vises i nettleser-fanen.
+
+Vi bruker den innebygde hooken `useEffect` for å kjøre denne side-effekten inni komponenten vår. `useEffect` tar i mot en funksjon som skal utføre side-effektene for oss. Vi kan implementere det slik:
+
+```js
+React.useEffect(() => {
+  document.title = 'Min nye tittel';
+});
+```
+
+I vårt tilfelle vil vi at tekst-strengen skal gjenspeile hvilken bruker som har lastet opp bildet. Det er ikke noe vanskeligere enn vanlig:
+
+```js
+React.useEffect(() => {
+  document.title = `📷 av ${image.user}`;
+});
+```
+
+Når du navigerer fra ett bilde til et annet ser du at tittelen oppdaterer seg. Du må riktignok åpne panelet til høyre i ene egen fane for å se det.
```

Comment:
Ny fane gjelder kun for de som sitter i CodeSandbox, endre til "For de som jobber i CodeSandbox må riktignok åpne visningspanelet til høyre i en egen fane for å se det"